plugins/org.wso2.developerstudio.eclipse.gmf.esb/src/org/wso2/developerstudio/eclipse/gmf/esb/CacheProtocolType.java

```diff
@@ -0,0 +1,200 @@
+/**
+ * Copyright 2009-2012 WSO2, Inc. (http://wso2.com)
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.developerstudio.eclipse.gmf.esb;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.eclipse.emf.common.util.Enumerator;
+
+/**
+ * <!-- begin-user-doc -->
+ * A representation of the literals of the enumeration '<em><b>Cache Protocol Type</b></em>',
+ * and utility methods for working with them.
+ * <!-- end-user-doc -->
+ * @see org.wso2.developerstudio.eclipse.gmf.esb.EsbPackage#getCacheProtocolType()
+ * @model
+ * @generated
+ */
+public enum CacheProtocolType implements Enumerator {
+	/**
+	 * The '<em><b>HTTP</b></em>' literal object.
+	 * <!-- begin-user-doc -->
+	 * <!-- end-user-doc -->
+	 * @see #HTTP_VALUE
+	 * @generated
+	 * @ordered
+	 */
+	HTTP(0, "HTTP", "HTTP");
+
+	/**
+	 * The '<em><b>HTTP</b></em>' literal value.
+	 * <!-- begin-user-doc -->
+	 * <p>
+	 * If the meaning of '<em><b>HTTP</b></em>' literal object isn't clear,
+	 * there really should be more of a description here...
+	 * </p>
+	 * <!-- end-user-doc -->
+	 * @see #HTTP
+	 * @model
+	 * @generated
+	 * @ordered
+	 */
+	public static final int HTTP_VALUE = 0;
+
+	/**
+	 * An array of all the '<em><b>Cache Protocol Type</b></em>' enumerators.
+	 * <!-- begin-user-doc -->
+	 * <!-- end-user-doc -->
+	 * @generated
+	 */
+	private static final CacheProtocolType[] VALUES_ARRAY =
+		new CacheProtocolType[] {
+			HTTP,
+		};
+
+	/**
+	 * A public read-only list of all the '<em><b>Cache Protocol Type</b></em>' enumerators.
+	 * <!-- begin-user-doc -->
+	 * <!-- end-user-doc -->
+	 * @generated
+	 */
+	public static final List<CacheProtocolType> VALUES = Collections.unmodifiableList(Arrays.asList(VALUES_ARRAY));
+
+	/**
+	 * Returns the '<em><b>Cache Protocol Type</b></em>' literal with the specified literal value.
+	 * <!-- begin-user-doc -->
+	 * <!-- end-user-doc -->
+	 * @param literal the literal.
+	 * @return the matching enumerator or <code>null</code>.
+	 * @generated
+	 */
+	public static CacheProtocolType get(String literal) {
+		for (int i = 0; i < VALUES_ARRAY.length; ++i) {
+			CacheProtocolType result = VALUES_ARRAY[i];
+			if (result.toString().equals(literal)) {
+				return result;
+			}
+		}
```

Comment:
We can use for each loop here;
for(CacheProtocolType result : VALUES_ARRAY) {
if (result.toString().equals(literal)) {
return result;
}
}